### ISSUE TITLE
Use TagGroup ffrom react-aria-components for filters #329

### ISSUE THREAD

**Issue by radoslavzeman** [bratislava/mestskakniznica.sk#283]
Filter tags on search page should be implemented using new TagGroup components from `react-aria-components` to ensure accessibility.

![image](https://github.com/bratislava/mestskakniznica.sk/assets/19418224/daceafb5-fca8-48f4-9122-177f903afcef)

Check bratislava.sk project, article filters.

Same as marianum: https://github.com/bratislava/marianum.sk/issues/329
